### ISSUE TITLE
Fix last.fm scrobbling behaviour and support now playing

### ISSUE THREAD

**Issue by shanedabes** [owntone/owntone-server#1321]
It seems like owntone isn't scrobbling as last.fm suggests. A track is only scrobbled once the full track has been played, but according to [this section of the documentation](https://www.last.fm/api/scrobbling#when-is-a-scrobble-a-scrobble):

> the track has been played for at least half its duration, or for 4 minutes (whichever occurs earlier.)

Also, it would be good to support the now playing functionality, which gets sent to last.fm as soon as a track begins to play. It is described [here](https://www.last.fm/api/scrobbling#now-playing-requests) and the api call is [here](https://www.last.fm/api/show/track.updateNowPlaying).

Thanks!

**Comment by ejurgensen** [comment]
Sounds fine. I'll add the "help wanted" tag since I probably won't get around to this myself.

**Comment by dreikelvin** [comment]
Hi there! I just installed linuxserver/daapd (aka "Owntone") on my synology NAS via docker and it seems like Last.fm scrobbling is not working at all. None of the tracks I play - either via Apple Music or in the web frontent are showing up in my Last.fm profile. I assume that this issue has been known since Oct 2021 and nobody has gotten around fixing it :P I don't know C either, sorry. Just what an Oscillator is and how you fix phasing issues in a mix haha

**Comment by ejurgensen** [comment]
> At least not if you just play a quick section of it

OwnTone scrobbles when the entire track has been played. Only if it's played by OwnTone, so not Apple Music. I think that should work, I tested it not too long ago.

**Comment by dreikelvin** [comment]
You're right! I just tested it in Apple Music and it doesn't scrobble at all. Any way to fix this? I was assuming that Apple Music basically access the same api as the owntone web interface.

**Comment by ejurgensen** [comment]
Now I actually looked at the code, and I see scrobbling is supposed to work also with Apple Music. I'll have to test that to see what the issue might be.
